src/Microsoft.PowerShell.Commands.Utility/commands/utility/AddType.cs

```diff
@@ -1090,36 +1209,173 @@ private void CompileSourceToAssembly(string source)
             }
             else
             {
-                emitResult = compilation.Emit(outputAssembly);
+                using (var fs = new FileStream(_outputAssembly, FileMode.CreateNew, FileAccess.ReadWrite, FileShare.None))
+                {
+                    emitResult = compilation.Emit(peStream: fs, options: emitOptions);
+                }
+
+                HandleCompilerErrors(emitResult.Diagnostics);
+
                 if (emitResult.Success)
                 {
+                    Assembly assembly = Assembly.LoadFrom(_outputAssembly);
```

Comment:
You cannot load the assembly unconditionally. Only load the assembly when `-PassThru` is specifed.

Reply:
Fixed.